docs/synapse/userguides/storm_ref_filter.rstorm

```diff
@@ -793,247 +917,449 @@ know the exact order or values of the array itself.
   details on working with arrays.
 
+
+.. _tag-filter:
+
+Tag Filters
+-----------
+
+Tags in Synapse can represent observations or assessments. They are used to provide context to nodes (in the
+form of "labels" applied to nodes) and to group related nodes.
+
+Storm supports filtering nodes based on the tags applied to nodes (including the use of tag globs), as well
+as filtering based on tag timestamps, tag properties, or tag property values.
+
+The "hashtag" symbol ( ``#`` ) is used to specify a tag name when filtering by tag.
+
 .. _filter-tag:
 
 Filter by Tag (#)
 +++++++++++++++++
 
-The tag extended comparator (``#``) supports filtering nodes based on the tags applied to a node.
-You can filter based on a given tag or on the timestamps associated with a given tag (using the
-interval comparator, ``@=``).
+A "filter by tag" operation downselects the current working set to include (or exclude) all nodes with
+the specified tag.
 
-.. NOTE::
+**Syntax:**
+
+*<query>* **+** | **-** **#** *<tag>*
+
+*Filter the current working set to exclude all nodes that ESET associates with Sednit:*
+
+.. storm-pre:: [ inet:fqdn=kg-news.org inet:ipv4=92.114.92.125 +#rep.eset.sednit ] -#rep.eset.sednit
+::
+  
+  <query> -#rep.eset.sednit
+
+*Filter the current working set to only include nodes associated with anonymized infrastructure:*
+
+.. storm-pre:: [ (inet:fqdn=ca2.vpn.airdns.org +#cno.infra.anon.vpn) (inet:ipv4=104.244.73.193 +#cno.infra.anon.tor.exit) ] +#cno.infra.anon
+::
   
-  You can also use "filter by tag" to filter nodes based on any tags with tag properties, or tags
-  with tag properties and specific tag property values, if tag properties are present in your data
-  model.
+  <query> +#cno.infra.anon
+
+.. TIP::
   
-  :ref:`tag-properties` are still supported by Synapse, but their use has largely been deprecated
-  in favor of extended model properties where needed.
+  Tags are hierarchical, and each tag element is its own tag; the tag ``#cno.infra.anon`` consists of the tags ``#cno``,
+  ``#cno.infra``, and ``#cno.infra.anon``. Filtering nodes using a tag "higher up" in the tag hierarchy will include
+  (or exclude) nodes with specified tag or any tag "lower down" in the hierarchy. In other words, filtering by ``#cno.infra.anon``
+  will filter all "anonymized" infrastructure, whether the infrastructure is a VPN (``#cno.infra.anon.vpn``), a TOR node
+  (``#cno.infra.anon.tor``), or an anonymous proxy (``#cno.infra.anon.proxy``).
+
+
+.. _filter-tag-globs:
+
+Filter by Tag Globs
++++++++++++++++++++
+
+Synapse supports filtering based on the set of tags that match a specified glob expression using single ( ``*`` )
+or double ( ``**`` ) asterisks, or a combination of the two.
+
+The single asterisk and double asterisk behave differently:
+
+- The asterisk ( ``*`` ) represents an arbitrary string that matches a single tag element (i.e., one element
```

Comment:
A single asterisk can also do partial matches of a single tag element, like `+#test.foo*.tag` would match `#test.foobar.tag`

Reply:
TIL. Clarified and added example.